### PR TITLE
Generate documentation as part of CI build.

### PR DESCRIPTION
**Pull request by mmphego**
This PR adds the following:
- Generates documentation as part of Jenkins CI build.
- Minor documentation fixes.
- Updated Readme file to include badges.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -7,15 +7,27 @@
...
+
 long_description = ""
 for name, filename in files.items():
-    long_description += "{}\n\n".format(name)
+    long_description += "# {}\n\n".format(name)
     with open(path.join(this_directory, filename)) as f:
         file_contents = f.read()
     long_description += file_contents + "\n\n"
 
+long_description = ""
```
Comment (ajoubertza) [inline setup.py:20]: This overwrites the value built up by the previous loop?

File: setup.py [hunk 2]
```diff
@@ -7,15 +7,27 @@
...
+for name, filename in files.items():
+    if name != 'Readme':
+        long_description += "# {}\n".format(name)
+    with open(path.join(this_directory, filename)) as _f:
+        file_contents = _f.read()
+    long_description += file_contents + "\n\n"
+
+
+
+print(long_description)
```
Comment (ajoubertza) [inline setup.py:30]: Is this still WIP?

File: setup.py (hunk 1 above)
Comment (mmphego) [reply setup.py]: Duplicated code.

File: setup.py (hunk 2 above)
Comment (mmphego) [reply setup.py]: Nope ready for review.

**Pull request merged**